x-pack/plugins/cases/server/services/cases/index.ts

```diff
@@ -299,48 +211,21 @@ export class CasesService {
   public async findCasesGroupedByID({
     unsecuredSavedObjectsClient,
     caseOptions,
-    subCaseOptions,
   }: {
     unsecuredSavedObjectsClient: SavedObjectsClientContract;
     caseOptions: FindCaseOptions;
-    subCaseOptions?: SavedObjectFindOptionsKueryNode;
   }): Promise<CasesMapWithPageInfo> {
     const cases = await this.findCases({
       unsecuredSavedObjectsClient,
       options: caseOptions,
     });
 
-    const subCasesResp = ENABLE_CASE_CONNECTOR
-      ? await this.findSubCasesGroupByCase({
-          unsecuredSavedObjectsClient,
-          options: subCaseOptions,
-          ids: cases.saved_objects
-            .filter((caseInfo) => caseInfo.attributes.type === CaseType.collection)
-            .map((caseInfo) => caseInfo.id),
-        })
-      : { subCasesMap: new Map<string, SubCaseResponse[]>(), page: 0, perPage: 0 };
-
     const casesMap = cases.saved_objects.reduce((accMap, caseInfo) => {
-      const subCasesForCase = subCasesResp.subCasesMap.get(caseInfo.id);
-
-      /**
-       * If this case is an individual add it to the return map
-       * If it is a collection and it has sub cases add it to the return map
-       * If it is a collection and it does not have sub cases, check and see if we're filtering on a status,
-       *  if we're filtering on a status then exclude the empty collection from the results
-       *  if we're not filtering on a status then include the empty collection (that way we can display all the collections
-       *  when the UI isn't doing any filtering)
-       */
-      if (
-        caseInfo.attributes.type === CaseType.individual ||
-        subCasesForCase !== undefined ||
-        !caseOptions.status
-      ) {
-        accMap.set(caseInfo.id, { case: caseInfo, subCases: subCasesForCase });
-      }
+      accMap.set(caseInfo.id, { case: caseInfo });
       return accMap;
     }, new Map<string, Collection>());
 
+    // TODO: use an aggregation here I think
```

Comment:
I think it better to do it on another PR. Relevant issue: https://github.com/elastic/kibana/issues/101995

Reply:
Also added it here: https://github.com/elastic/kibana/issues/123759